okhttp/src/main/java/okhttp3/ResponseBody.kt

```diff
@@ -221,36 +221,84 @@ abstract class ResponseBody : Closeable {
           charset = resolvedCharset
         }
       }
-      val buffer = Buffer().writeString(content, charset)
-      return create(finalContentType, buffer.size, buffer)
+      val buffer = Buffer().writeString(this, charset)
+      return buffer.toResponseBody(finalContentType, buffer.size)
     }
 
-    /** Returns a new response body that transmits [content]. */
+    /** Returns a new response body that transmits this byte array. */
     @JvmStatic
-    fun create(contentType: MediaType?, content: ByteArray): ResponseBody {
-      val buffer = Buffer().write(content)
-      return create(contentType, content.size.toLong(), buffer)
+    @JvmName("create")
+    fun ByteArray.toResponseBody(contentType: MediaType? = null): ResponseBody {
+      return Buffer()
+          .write(this)
+          .toResponseBody(contentType, size.toLong())
     }
 
-    /** Returns a new response body that transmits [content]. */
+    /** Returns a new response body that transmits this byte string. */
     @JvmStatic
-    fun create(contentType: MediaType?, content: ByteString): ResponseBody {
-      val buffer = Buffer().write(content)
-      return create(contentType, content.size.toLong(), buffer)
+    @JvmName("create")
+    fun ByteString.toResponseBody(contentType: MediaType? = null): ResponseBody {
+      return Buffer()
```

Comment:
Seems like we should be able to avoid the copy into new Buffer, since the source is an immutable ByteArray wrapper.

Reply:
The catch is this ultimately needs to produce a BufferedSource, so copying into a Buffer is inescapable.

The other catch is the source has to be one-shot, so if we defer that work we have to do some cursor management.